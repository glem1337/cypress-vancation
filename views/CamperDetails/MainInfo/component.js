import React, { memo } from 'react';
import { FormattedMessage } from 'react-intl';
import { Tag } from 'antd';
import * as R from 'ramda';
import classnames from 'classnames';

import isPresent from 'utils/isPresent';
import SkeletonTitle from 'views/shared/SkeletonTitle';

import useContainer from './hook';

const MainInfo = () => {
  const {
    camper,
    milesCount,
    deliveryDistance,
    ratingColors,
    isOffGridCapableExist,
  } = useContainer();

  if (!isPresent(camper)) {
    return (
      <div className="van-details__name-wrap van-details__name-wrap--skeleton">
        <SkeletonTitle />
        <SkeletonTitle />
        <SkeletonTitle width={60} />
        <SkeletonTitle className="mb-0" width={35} />
      </div>
    );
  }

  return (
    <div className="van-details__name-wrap">
      <h1 className="main-title mb-16">
        {camper?.name}
      </h1>
      <div className="van-details__tags-wrap">
        {camper?.glamper && (
          <div className="tag-glamper mr-8">
            <img
              className="mr-4"
              src="/images/profile/Diamond-White.svg"
              alt=""
            />
            <FormattedMessage id="shared.glamper" />
          </div>
        )}
        {deliveryDistance > 0 && (
          <Tag
            className="mr-8"
            color="default"
            icon={<i className="icon icon-delivery in-black" />}
          >
            <span className="in-black">
              <FormattedMessage
                id="camperDetails.mainInfo.tag.delivery"
                values={{ miles: deliveryDistance }}
              />
            </span>
          </Tag>
        )}
        {isOffGridCapableExist && (
          <Tag
            className="mr-8"
            color="default"
            icon={(
              <img
                className="mr-4"
                src="/images/details_camp/Off-Grid Capable.svg"
                alt=""
              />
          )}
          >
            <span className="in-black">
              <FormattedMessage id="camperDetails.mainInfo.tag.offGridCapable" />
            </span>
          </Tag>
        )}
        <Tag
          className="mr-8"
          color="default"
          icon={(
            <img
              className="van-details__miles-tag"
              src="/images/edit_listing/rules/Allow-Unlimited-Miles.svg"
              alt=""
            />
          )}
        >
          {isPresent(milesCount)
            ? (
              <span className="in-black">
                <FormattedMessage
                  id="camperDetails.mainInfo.tag.limitedMilesCount"
                  values={{ miles: milesCount }}
                />
              </span>
            )
            : (
              <span className="in-black">
                <FormattedMessage id="camperDetails.mainInfo.tag.unlimitedMilesFree" />
              </span>
            )}
        </Tag>
      </div>
      <div className="d-flex align-items-center">
        <div
          className={
            classnames(
              'profile-user-card__photo-rating-icon',
              ratingColors.background,
            )
          }
        >
          <img src="/images/Like - White.svg" alt="" />
        </div>
        <p
          className={
            classnames(
              'mr-16 in-green-300 text-title',
              ratingColors.text,
            )
          }
        >
          {`${camper?.raiting}%`}
        </p>
      </div>
    </div>
  );
};

export default memo(MainInfo, R.T);
