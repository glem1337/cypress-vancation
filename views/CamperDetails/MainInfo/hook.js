import { useRouter } from 'next/router';
import { useSelector } from 'react-redux';
import * as R from 'ramda';

import { camperSelector } from 'state/concepts/camper/selectors';
import { RATING_BOUNDARIES } from 'constants/camperDetails/owner';

function useContainer() {
  const router = useRouter();

  const camper = useSelector(state => camperSelector(
    state,
    router.query.camper_id,
  ));

  // Miles
  const milesLimited = R.path(['tripFee', 'tripFeeMileage', 'limit'], camper) === true;
  const milesCount = milesLimited
    ? R.path(['tripFee', 'tripFeeMileage', 'available'], camper)
    : null;

  // Delivery
  const deliveryDistance = R.path(['deliveryInformation', 'distance'], camper);

  // Rating color
  const rating = camper?.raiting;

  const ratingColors = R.cond([
    [
      value => value >= RATING_BOUNDARIES.HIGH,
      R.always({
        text: 'in-green-1000',
        background: 'background-green-300',
      }),
    ],
    [
      value => value >= RATING_BOUNDARIES.NORMAL.START && value <= RATING_BOUNDARIES.NORMAL.END,
      R.always({
        text: 'in-green-300',
        background: 'background-green-300',
      }),
    ],
    [
      value => value >= RATING_BOUNDARIES.MIDDLE.START && value <= RATING_BOUNDARIES.MIDDLE.END,
      R.always({
        text: 'in-yellow-1000',
        background: 'background-yellow-1000',
      }),
    ],
    [
      R.T,
      R.always({
        text: 'in-red-1000',
        background: 'background-red-1000',
      }),
    ],
  ])(rating);

  /**
   * Detect Off grid capable system.
   */
  const detectOffGridCapableSystem = () => {
    const amenities = R.defaultTo([], camper?.amenities);

    const CONFIGURATION_AMENITY_TITLE = 'Power System';
    const CONFIGURATION_AMENITY_OPTION_TITLE = 'Off Grid Capable System';

    for (let i = 0; i < amenities.length; i += 1) {
      const amenity = amenities[i];

      const configurationAmenityTitle = R.path(['configurationAmenity', 'title'], amenity);

      if (configurationAmenityTitle === CONFIGURATION_AMENITY_TITLE) {
        const options = R.pathOr([], ['amenityOptions'], amenity);

        for (let j = 0; j < options.length; j += 1) {
          const option = options[j];

          const title = R.path(['configurationAmenityOption', 'title'], option);

          if (title === CONFIGURATION_AMENITY_OPTION_TITLE) {
            return true;
          }
        }
      }
    }

    return false;
  };

  return {
    camper,
    milesCount,
    milesLimited,
    deliveryDistance,
    ratingColors,
    isOffGridCapableExist: detectOffGridCapableSystem(),
  };
}

export default useContainer;
