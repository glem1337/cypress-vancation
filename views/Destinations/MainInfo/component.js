import React from 'react';
import classNames from 'classnames';

import isPresent from 'utils/isPresent';

import InnerHtmlField from '../InnerHtmlField';

import useContainer from './hook';

const MainInfo = () => {
  const { title, subTitle } = useContainer();

  return (
    <>
      <h1 className={classNames('text-headline', { 'd-none': !isPresent(title) })}>
        <InnerHtmlField html={title} />
      </h1>
      <p className={classNames('search-page__subtitle-info', { 'd-none': !isPresent(subTitle) })}>
        <InnerHtmlField html={subTitle} />
      </p>
    </>
  );
};

export default MainInfo;
