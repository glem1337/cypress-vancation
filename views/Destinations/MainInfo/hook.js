import { useSelector } from 'react-redux';
import { path } from 'ramda';

import { searchDestinationParamsSelector } from 'state/concepts/search-destinations/selectors';

function useContainer() {
  const searchParams = useSelector(searchDestinationParamsSelector);

  return {
    title: path(['location', 'seoInfo', 'pageHeaderTitle'], searchParams),
    subTitle: path(['location', 'subtitle'], searchParams),
  };
}

export default useContainer;
