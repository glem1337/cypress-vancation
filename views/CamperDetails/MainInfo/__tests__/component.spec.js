import React from 'react';
import { shallow } from 'enzyme';

import MainInfo from '../component';

let mockedHookData = {
  camper: {
    name: 'Super camper name',
    glamper: true,
    raiting: 50,
  },
  milesCount: null,
  deliveryDistance: 123,
  ratingColors: {
    background: 'in-red-1000',
    text: 'in-red-1000',
  },
};
jest.mock('../hook', () => jest.fn(() => mockedHookData));

describe('MainInfo component tests', () => {
  it('matches snapshot', () => {
    const component = shallow(<MainInfo />);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when milesCount is not null', () => {
    mockedHookData = {
      ...mockedHookData,
      milesCount: 345,
    };

    const component = shallow(<MainInfo />);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when camper is not existing', () => {
    mockedHookData = {
      ...mockedHookData,
      camper: null,
    };

    const component = shallow(<MainInfo />);

    expect(component).toMatchSnapshot();
  });
});
