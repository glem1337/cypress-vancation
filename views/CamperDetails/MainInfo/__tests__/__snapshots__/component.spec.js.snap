// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainInfo component tests matches snapshot 1`] = `
<div
  className="van-details__name-wrap"
>
  <h1
    className="main-title mb-16"
  >
    Super camper name
  </h1>
  <div
    className="van-details__tags-wrap"
  >
    <div
      className="tag-glamper mr-8"
    >
      <img
        alt=""
        className="mr-4"
        src="/images/profile/Diamond-White.svg"
      />
      <Memo(FormattedMessage)
        id="shared.glamper"
      />
    </div>
    <Tag
      className="mr-8"
      color="default"
      icon={
        <i
          className="icon icon-delivery in-black"
        />
      }
    >
      <span
        className="in-black"
      >
        <Memo(FormattedMessage)
          id="camperDetails.mainInfo.tag.delivery"
          values={
            Object {
              "miles": 123,
            }
          }
        />
      </span>
    </Tag>
    <Tag
      className="mr-8"
      color="default"
      icon={
        <img
          alt=""
          className="van-details__miles-tag"
          src="/images/edit_listing/rules/Allow-Unlimited-Miles.svg"
        />
      }
    >
      <span
        className="in-black"
      >
        <Memo(FormattedMessage)
          id="camperDetails.mainInfo.tag.unlimitedMilesFree"
        />
      </span>
    </Tag>
  </div>
  <div
    className="d-flex align-items-center"
  >
    <div
      className="profile-user-card__photo-rating-icon in-red-1000"
    >
      <img
        alt=""
        src="/images/Like - White.svg"
      />
    </div>
    <p
      className="mr-16 in-green-300 text-title in-red-1000"
    >
      50%
    </p>
  </div>
</div>
`;

exports[`MainInfo component tests matches snapshot when camper is not existing 1`] = `
<div
  className="van-details__name-wrap van-details__name-wrap--skeleton"
>
  <SkeletonTitle
    className=""
    width={100}
  />
  <SkeletonTitle
    className=""
    width={100}
  />
  <SkeletonTitle
    className=""
    width={60}
  />
  <SkeletonTitle
    className="mb-0"
    width={35}
  />
</div>
`;

exports[`MainInfo component tests matches snapshot when milesCount is not null 1`] = `
<div
  className="van-details__name-wrap"
>
  <h1
    className="main-title mb-16"
  >
    Super camper name
  </h1>
  <div
    className="van-details__tags-wrap"
  >
    <div
      className="tag-glamper mr-8"
    >
      <img
        alt=""
        className="mr-4"
        src="/images/profile/Diamond-White.svg"
      />
      <Memo(FormattedMessage)
        id="shared.glamper"
      />
    </div>
    <Tag
      className="mr-8"
      color="default"
      icon={
        <i
          className="icon icon-delivery in-black"
        />
      }
    >
      <span
        className="in-black"
      >
        <Memo(FormattedMessage)
          id="camperDetails.mainInfo.tag.delivery"
          values={
            Object {
              "miles": 123,
            }
          }
        />
      </span>
    </Tag>
    <Tag
      className="mr-8"
      color="default"
      icon={
        <img
          alt=""
          className="van-details__miles-tag"
          src="/images/edit_listing/rules/Allow-Unlimited-Miles.svg"
        />
      }
    >
      <span
        className="in-black"
      >
        <Memo(FormattedMessage)
          id="camperDetails.mainInfo.tag.limitedMilesCount"
          values={
            Object {
              "miles": 345,
            }
          }
        />
      </span>
    </Tag>
  </div>
  <div
    className="d-flex align-items-center"
  >
    <div
      className="profile-user-card__photo-rating-icon in-red-1000"
    >
      <img
        alt=""
        src="/images/Like - White.svg"
      />
    </div>
    <p
      className="mr-16 in-green-300 text-title in-red-1000"
    >
      50%
    </p>
  </div>
</div>
`;
